solidity/contracts/system/TBTCSystem.sol

```diff
@@ -142,13 +143,32 @@ contract TBTCSystem is Ownable, ITBTCSystem, DepositLog {
             _keepThreshold,
             _keepSize
         );
+        vendingMachine = _vendingMachine;
         setTbtcDepositToken(_tbtcDepositToken);
-        _initialized = true;
+        initialized = true;
         allowNewDeposits = true;
     }
 
-    /// @notice gets whether new deposits are allowed.
-    function getAllowNewDeposits() external view returns (bool) { return allowNewDeposits; }
+    /// @notice returns whether new deposits should be allowed.
+    function getAllowNewDeposits() external view returns (bool) {
+        if (!allowNewDeposits) { return false; }
+
+        uint256 supply = vendingMachine.getMintedSupply();
+        uint256 bufferedSupply = supply.add(getMaxLotSize().mul(10 ** 10));
+
+        return bufferedSupply <= vendingMachine.getMaxSupply();
```

Comment:
We’re comfortable allowing no new deposits if the largest available lot would push us over the cap is the goal here right?

Reply:
Correct.